modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ParameterResolverValueResolverWrapper.java

```diff
@@ -6,26 +6,69 @@
  */
 package org.mule.runtime.module.extension.internal.runtime.resolver;
 
+import static java.util.Optional.empty;
+import static java.util.Optional.ofNullable;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.initialiseIfNeeded;
 import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.lifecycle.Initialisable;
+import org.mule.runtime.api.lifecycle.InitialisationException;
+import org.mule.runtime.api.util.Reference;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.context.MuleContextAware;
 import org.mule.runtime.core.api.event.BaseEvent;
 import org.mule.runtime.extension.api.runtime.parameter.ParameterResolver;
 
+import java.util.Optional;
+import java.util.function.Function;
+
 /**
- * {@link ValueResolver} implementation for {@link ParameterResolver} that are not resolved from an
- * expression.
- * <p>
- * This {@link ParameterResolverValueResolverWrapper} delegates the resolution to the given {@link ValueResolver} from
- * the constructor and wraps the value into a {@link StaticParameterResolver}
+ * {@link ValueResolver} implementation for {@link ParameterResolver} that wraps another {@link ValueResolver}
  *
  * @since 4.0
  * @see ParameterResolver
  */
-public class ParameterResolverValueResolverWrapper<T> implements ValueResolver<ParameterResolver<T>> {
+public class ParameterResolverValueResolverWrapper<T>
+    implements ValueResolver<ParameterResolver<T>>, Initialisable, MuleContextAware {
 
-  private ValueResolver resolver;
+  private ValueResolver<T> resolver;
+  private MuleContext muleContext;
+  private Reference<Function<ValueResolvingContext, ParameterResolver>> parameterResolverSupplier = new Reference<>();
 
   public ParameterResolverValueResolverWrapper(ValueResolver resolver) {
     this.resolver = resolver;
+    Function<ValueResolvingContext, ParameterResolver> parameterResolverFactory = (context) -> new ParameterResolver<T>() {
+
+      @Override
+      public T resolve() {
+        try {
+          Object resolve = resolver.resolve(context);
+          if (resolve instanceof ValueResolver) {
+            resolve = ((ValueResolver) resolve).resolve(context);
+          }
+          return (T) resolve;
+        } catch (MuleException e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+      @Override
+      public Optional<String> getExpression() {
+        return resolver instanceof ExpressionBasedValueResolver
+            ? ofNullable(((ExpressionBasedValueResolver) resolver).getExpression()) : empty();
+      }
+    };
+
+    if (resolver.isDynamic()) {
+      parameterResolverSupplier.set(parameterResolverFactory);
+    } else {
+      parameterResolverSupplier.set(context -> {
+        ParameterResolver staticResolver = parameterResolverFactory.apply(context);
+        Function<ValueResolvingContext, ParameterResolver> valueResolvingContextParameterResolverFunction =
```

Comment:
why do you need this in the static case?

Reply:
You mean to pass through the `ValueResolvingContext`?